vm/src/builtins/complex.rs

```diff
@@ -59,14 +60,31 @@ impl PyObjectRef {
         }
         if let Some(method) = vm.get_method(self.clone(), identifier!(vm, __complex__)) {
             let result = method?.call((), vm)?;
-            // TODO: returning strict subclasses of complex in __complex__ is deprecated
-            return match result.payload::<PyComplex>() {
-                Some(complex_obj) => Ok(Some((complex_obj.value, true))),
-                None => Err(vm.new_type_error(format!(
-                    "__complex__ returned non-complex (type '{}')",
-                    result.class().name()
-                ))),
-            };
+
+            let ret_class = result.class().to_owned();
+            if let Some(ret) = result.downcast_ref::<PyComplex>() {
+                warnings::warn(
+                    vm.ctx.exceptions.deprecation_warning,
+                    format!(
+                        "__complex__ returned non-complex (type {}).  \
+                    The ability to return an instance of a strict subclass of complex \
+                    is deprecated, and may be removed in a future version of Python.",
+                        ret_class
+                    ),
+                    1,
+                    vm,
+                )?;
+
+                Ok(Some((ret.value, true)))
```

Comment:
```suggestion
                return Ok(Some((ret.value, true)));
```

My mistake here, it thought it was the last line in the function.